### DISCUSSION TITLE
Apostrophe 3 and Multi-tenancy

### DISCUSSION THREAD

**Discussion by LoneWolf741** [apostrophecms/apostrophe#3821] [General]
I stumbled upon Apostrophe a few days ago and really like what I see. I'm looking for a CMS to house  a few client sites (and hoping to grow to more).  

All the client sites I host are similar and require the same tools.  Is Apostrophe 3 able to do Multi-tenancy out of the box? Everything I see for Multi-tenancy is mentioning Apostrophe 2 module.  

I understand there is Apostrophe Assembly - for a fleet of sites, but looking to self host a single code base.

I'm in the beginning stages of looking for a platform to learn and grow with so hoping apostrophe 3 (and future versions) has something built in for Multi-tenancy. 

Thanks in advance!

**Comment by boutell** [comment]
Hi @LoneWolf741, if by "multitenancy" you mean "many sites running on a single codebase, efficiently reusing apostrophe processes for multiple sites," etc. then yes this is exactly what Apostrophe Assembly is. The difference, from a technical standpoint, is that `@apostrophecms-pro/multisite`for 3.x  is a closed-source offering available as part of our Apostrophe Assembly license. And we also offer a highly scalable, available and durable hosted solution that allows you to just "git push" to deploy, although self-hosting is also possible with an Assembly license.

To start the conversation about Apostrophe Assembly for your project please reach out to hello@apostrophecms.com.

If you mean something else by "multitenancy" please clarify and we'll go from there.

**Reply by LoneWolf741** [reply]
Ok thank you for your response. I'm new-ish to programming, currently have a little bit of experience with Ruby/Rails, but wanting to pick a CMS to host a couple sites now and expand my knowledge. I am really starting to enjoy node, so Apos looked promising.  So I was really wanting to self host and "tinker and explore" with the option of hosting multiple sites under one code base as you describe, but on my server.

So from your above reply, I understand that what I want "multisite under one code base" was available as a module for Apos 2, but now with Apos 3, it is a paid service only.  Correct?

**Reply by boutell** [reply]
Well, if you're talking about a few sites, you can still certainly do this by deploying the same code base to several servers, or separate folders or containers on the same server with environment variables used to set things like the database name. Nobody's stopping you from doing that at all. You just don't get the very low overhead that comes with serving many sites out of a single running node process without the professional license.
